processing/src/main/java/io/druid/query/filter/BoundDimFilter.java

```diff
@@ -385,19 +385,20 @@ private void initLongData()
           if (hasLowerBound()) {
             final Long lowerLong = GuavaUtils.tryParseLong(lower);
             if (lowerLong == null) {
-              matchesAnything = false;
-              return;
+              // Unparseable values fall before all actual numbers, so all numbers will match the lower bound.
+              hasLowerLongBoundVolatile = false;
+            } else {
+              hasLowerLongBoundVolatile = true;
+              lowerLongBoundVolatile = lowerLong;
             }
-
-            hasLowerLongBoundVolatile = true;
-            lowerLongBoundVolatile = lowerLong;
           } else {
             hasLowerLongBoundVolatile = false;
```

Comment:
@himanshug I disagree with "why only one of them is marked volatile and not others but all are expected to behave like volatile". This is similar to a discussion below: https://github.com/druid-io/druid/pull/3797#discussion_r93733927. We don't try to make all other fields "to behave like volatile". Volatiles on all other fields in this anonymous `Supplier` implementation not only unnecessary, but also don't really provide any extra guarantees. E. g. leaving `volatile` on all fields, but removing it on `longsInitialized` would break this code. `volatile` on `longsInitialized` is necessary and sufficient.